tests/light/src/common/network.py

```diff
@@ -126,3 +126,49 @@ async def _start_server(self):
         server = await asyncio.start_unix_server(self._client_accepted_cb, self._path, ssl=self._ssl)
         await server.start_serving()
         return server
+
+
+class DatagramServer(ABC):
+    def __init__(self):
+        self._event_loop = BackgroundEventLoop()
+        self._sock = None
+
+    @abstractmethod
+    async def start(self):
+        pass
+
+    async def stop(self):
+        if self._sock is not None:
+            self._sock.close()
+            self._sock = None
+
+    async def read_dgram(self, maxsize=65536):
+        return await self._event_loop.loop.sock_recv(self._sock, maxsize)
+
+    async def write_dgram(self, data):
+        raise NotImplementedError
```

Comment:
`NotImplementedError` is a native Python built-in exception we do not need to implement is separately.

Reply:
I think the question is about the real implementation of `write_dgram()`.
I don't really remember why I did not implement this, it would be probably a single line:
```
return await self._event_loop.loop.sock_sendall(self._sock, data)
```

Reply:
Fixed